ui/src/components/Document/DocumentAuthors.js

```diff
@@ -1,28 +1,123 @@
-import { List } from 'semantic-ui-react';
+import { List, Popup, Icon } from 'semantic-ui-react';
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
 
 export class DocumentAuthors extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false,
+    };
+  }
+
+  renderPopupContent = author => {
+    return (
+      <>
+        {author.full_name}
+        <br />
+        {!isEmpty(author.roles) && (
```

Comment:
It would be nice to limit the checks for empty. Moreover, the list should be joined rather than using `toString`.
To achieve this, prefer components to functions (pseudocode):
```
class Author {
  render() {
    ...
    {author.roles.join(" ")}
    ...

  }
}

Author.propTypes = {
  author = Obj.shape(
    roles: array
  )
}
Author.default = {
  roles: []
}


class DocumentAuthors {
  render() {
    return authors.map((author, index) => <Author author={author} />
  }
}

DocumentAuthors.propTypes = {
    authors: array
}
DocumentAuthors.default = {
  authors: []
}

```
Like this you don't need less checks and you take advantage of default props. This is applicable to each field.

This is waht I was mentioning to @kprzerwa yesterday too.